app/src/main/java/com/realityexpander/tasky/agenda_feature/common/RepositoryResult.kt

```diff
@@ -1,6 +1,15 @@
 package com.realityexpander.tasky.agenda_feature.common
 
-sealed interface RepositoryResult {
-    object Success : RepositoryResult
-    data class Error(val message: String) : RepositoryResult
+import com.realityexpander.tasky.core.presentation.common.util.UiText
+
+sealed interface RepositoryResult<T> {
+    data class Success<T>(
+        val data: T? = null,
+        val message: String? = null
+    ) : RepositoryResult<T>
+
+    data class Error<T>(
+        val message: UiText,
+        val originalMessage: String? = null
+    ) : RepositoryResult<T>
 }
```

Comment:
I'd suggest calling this something a bit more generic, since something to distinguish between success and error might be used for more than just in repositories

Reply:
OK, sounds good.